### PR TITLE
Add libssh2/1.9.0

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **libssh2/1.9.0**

Also updated for conventions.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libssh2/all/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.8.12)
+cmake_minimum_required(VERSION 2.8.12)
```
Comment (uilianries) [inline recipes/libssh2/all/CMakeLists.txt]: @theirix did you change it because a hook error? If yes, it's a bug.

Comment (theirix) [reply recipes/libssh2/all/CMakeLists.txt]: yes, the hook does not check for whitespace

Comment (Croydon) [reply recipes/libssh2/all/CMakeLists.txt]: Is it valid CMake to have there whitespaces? 🤔

Comment (theirix) [reply recipes/libssh2/all/CMakeLists.txt]: Yes, it matches grammar and works well. Also it workes even with such ancient CMake version.

Comment (uilianries) [reply recipes/libssh2/all/CMakeLists.txt]: okay, I'll send a patch to fix it.

**Pull request merged**